test/test_compile.py

```diff
@@ -0,0 +1,97 @@
+import argparse
+import os.path
+import pytest
+
+from otk.command import compile
+
+
+@pytest.mark.parametrize(
+    "parser, arguments, dry_run, input_data, output_data, sys_exit_code, log_message",
+    [
+        # "GENERATE" is a placeholder to append "tmp_path" from pytest
+        # dry_run must not generate an output
+        (
+            None,
+            argparse.Namespace(input="GENERATE", output="GENERATE", target=None),
+            True,
+            """otk.version: 1
+otk.target.osbuild.qcow2: { test: 1 }
+""",
+            None,
+            0,
+            None,
+        ),
+        (
+            None,
+            argparse.Namespace(input="GENERATE", output=None, target=None),
+            True,
+            """otk.version: 1
+otk.target.osbuild.qcow2: { test: 1 }
+""",
+            None,
+            0,
+            None,
+        ),
+        (
+            None,
+            argparse.Namespace(input="GENERATE", output="GENERATE", target=None),
+            True,
+            """otk.version: 1
+otk.target.osbuild.qcow2: { test: 1 }
+otk.target.osbuild.ami: { test: 2 }
+""",
+            None,
+            1,
+            "INPUT contains multiple targets, `-t` is required",
+        ),
+        (
+            None,
+            argparse.Namespace(input="GENERATE", output="GENERATE", target=None),
+            False,
+            """otk.version: 1
+otk.target.osbuild.qcow2: { test: 1 }
+""",
+            """{
+  "test": 1,
+  "version": "2",
+  "sources": {}
+}""",
+            0,
+            None,
+        ),
+    ],
+)
+def test_validate(
```

Comment:
This is called "test_validate" but the test is called test_compile.py and the function under test is "compile()" too, is this intentional?

Reply:
Nope - just started with validate tests and forgot to rename when I saw we could as well test both